### PR TITLE
Elastic impl

### PR DESCRIPTION
**Pull request by DanialErfanian**

### CODE REVIEW

File: src/main/java/in/nimbo/exception/ElasticException.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package in.nimbo.exception;
+
+public class ElasticException extends Exception {
```
Comment (Borjianamin98) [inline src/main/java/in/nimbo/exception/ElasticException.java]: extend RuntimeException instead of exception for being unchecked

Comment (DanialErfanian) [reply src/main/java/in/nimbo/exception/ElasticException.java]: Fixed

File: src/main/java/in/nimbo/config/ElasticConfig.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
+package in.nimbo.config;
+
+import in.nimbo.exception.LoadConfigurationException;
+import org.apache.commons.configuration2.Configuration;
+import org.apache.commons.configuration2.builder.fluent.Configurations;
+import org.apache.commons.configuration2.ex.ConfigurationException;
+
+public class ElasticConfig {
+    private static final String CONFIG_NAME = "elastic-search.properties";
+    private String indexName, host, type;
```
Comment (Borjianamin98) [inline src/main/java/in/nimbo/config/ElasticConfig.java]: write properties in new lines

File: src/main/java/in/nimbo/dao/elastic/ElasticCLI.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+                case "getAllLinks":
+                    System.out.println(elasticDAO.getAllLinks());
+                    break;
+                case "get": {
+                    String link = scanner.next();
+                    System.out.println(elasticDAO.get(link));
+                    break;
+                }
+                case "save": {
+                    String link = scanner.next(), text = scanner.next();
```
Comment (Borjianamin98) [inline src/main/java/in/nimbo/dao/elastic/ElasticCLI.java]: write each in separate lines

File: src/main/java/in/nimbo/dao/elastic/ElasticDAOImpl.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        try {
+            GetRequest getRequest = new GetRequest(config.getIndexName(), config.getType(), link);
+            GetResponse response = client.get(getRequest, RequestOptions.DEFAULT);
+            if (response.isExists()) {
+                String text = (String) response.getSource().get("text");
+                return Optional.ofNullable(text);
+            }
+        } catch (IOException e) {
+            throw new ElasticException("Get failed", e);
+        } catch (ClassCastException ignored) {
```
Comment (Borjianamin98) [inline src/main/java/in/nimbo/dao/elastic/ElasticDAOImpl.java]: Do not use empty catch block.
Use optional.nullable instead

File: src/main/java/in/nimbo/exception/ElasticException.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+    public ElasticException(String message, Throwable cause) {
+        super(message, cause);
+    }
+
+    public ElasticException(Throwable cause) {
+        super(cause);
+    }
+
+    protected ElasticException(String message, Throwable cause, boolean enableSuppression, boolean writableStackTrace) {
```
Comment (Borjianamin98) [inline src/main/java/in/nimbo/exception/ElasticException.java]: Remove it

File: src/main/java/in/nimbo/config/ElasticConfig.java (hunk 2 above)
Comment (DanialErfanian) [reply src/main/java/in/nimbo/config/ElasticConfig.java]: done

**Pull request merged**